boto/https_connection.py

```diff
@@ -125,10 +125,28 @@ def connect(self):
         else:
             msg += "using system provided SSL certs"
         boto.log.debug(msg)
-        self.sock = ssl.wrap_socket(sock, keyfile=self.key_file,
-                                    certfile=self.cert_file,
-                                    cert_reqs=ssl.CERT_REQUIRED,
-                                    ca_certs=self.ca_certs)
+        if hasattr(ssl, 'SSLContext') and getattr(ssl, 'HAS_SNI', False):
+            # Use SSLContext so we can specify server_hostname for SNI
+            # (Required for connections to storage.googleapis.com)
```

Comment:
Have you tested what this does to requests sent to S3?

I want to make sure changes made for the sake of GCS functionality don't break S3 customer calls.

Reply:
Yep.  I ran the integration tests against S3; the tests were run with v1.1.0 of OpenSSL, which, according to the interpreter session below, should hit this code path:

```
>>> import ssl
>>> hasattr(ssl, 'SSLContext')
True
>>> getattr(ssl, 'HAS_SNI', False)
True
>>> ssl._OPENSSL_API_VERSION
(1, 1, 0, 6, 15)
```